### PR TITLE
Fix Issues

### PR DESCRIPTION
**Pull request by jvc56**

### CODE REVIEW

File: core/src/config.c [hunk 1]
```diff
@@ -369,69 +377,70 @@ void destroy_config(Config *config) {
...
-    strcpy(leaves, "data/lexica/french.klv2");
+    string_copy(leaves, "data/lexica/french.klv2");
   }
 
-  if (*config == NULL) {
-    *config = create_config(dist, cgp, lexicon_file, leaves, SORT_BY_EQUITY,
-                            PLAY_RECORDER_TYPE_ALL, "", "", SORT_BY_EQUITY,
-                            PLAY_RECORDER_TYPE_ALL, 0, 0, 9, 0, "", 0, 0, 0, 0,
-                            0, winpct, 100);
+  if (!*config) {
```
Comment (domino14) [inline core/src/config.c:396]: might it be better to be specific here? NULL is more specific (even though it is 0, yes)

File: core/src/gcg.c [hunk 2]
```diff
@@ -561,15 +562,15 @@ gcg_parse_status_t parse_next_gcg_line(GCGParser *gcg_parser) {
...
       token == GCG_EXCHANGE_TOKEN) {
-    if (game_history->players[0] == NULL || game_history->players[1] == NULL) {
+    if (!game_history->players[0] || !game_history->players[1]) {
       return GCG_PARSE_STATUS_MOVE_BEFORE_PLAYER;
     }
-    if (game_history->letter_distribution == NULL) {
+    if (!game_history->letter_distribution) {
 
-      if (game_history->lexicon_name == NULL) {
+      if (!game_history->lexicon_name) {
```
Comment (domino14) [inline core/src/gcg.c:570]: same NULL v 0 question here

File: core/src/klv.c [hunk 3]
```diff
@@ -93,27 +80,24 @@ float convert_little_endian_to_host(float little_endian_float) {
 
 void load_klv(KLV *klv, const char *klv_filename) {
   FILE *stream = stream_from_filename(klv_filename);
-  if (stream == NULL) {
-    perror(klv_filename);
-    exit(EXIT_FAILURE);
+  if (!stream) {
+    log_fatal("failred to open stream from filename: %s\n", klv_filename);
```
Comment (domino14) [inline core/src/klv.c:84]: `failed`

**Review by domino14 — COMMENTED** [review]
looks good just a q

File: core/src/config.c (hunk 1 above)
Comment (jvc56) [reply core/src/config.c:396]: Leaving as is for now, might change later.

File: core/src/gcg.c (hunk 2 above)
Comment (jvc56) [reply core/src/gcg.c:570]: Leaving as is for now, might change later.

File: core/src/klv.c (hunk 3 above)
Comment (jvc56) [reply core/src/klv.c:84]: Fixed in both cases.

**Pull request merged**